envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/EnvoyControlSynchronizationTest.kt

```diff
@@ -1,124 +1,155 @@
+            "envoy-control.sync.enabled" to true,
+            "envoy-control.sync.polling-interval" to pollingInterval.seconds
+        )
 
-    abstract val pollingInterval: Duration
-    abstract val stateSampleDuration: Duration
+        @JvmField
+        @RegisterExtension
+        val envoyControlDc1 = EnvoyControlClusteredExtension(consulClusters.serverFirst, { properties }, listOf(consulClusters))
 
-    private val logger by logger()
+        @JvmField
+        @RegisterExtension
+        val envoyControlDc2 = EnvoyControlClusteredExtension(consulClusters.serverSecond, dependencies = listOf(consulClusters))
+
+        @JvmField
+        @RegisterExtension
+        val envoy = EnvoyExtension(envoyControlDc1)
+
+        @JvmField
+        @RegisterExtension
+        val serviceLocal = EchoServiceExtension()
+
+        @JvmField
+        @RegisterExtension
+        val serviceRemote = EchoServiceExtension()
+    }
 
     @Test
     fun `should prefer services from local dc and fallback to remote dc when needed`() {
+
         // given: local and remote instances
-        registerServiceInRemoteCluster("echo", echoContainer2)
-        val localId = registerServiceInLocalDc("echo")
+        registerServiceInRemoteDc("echo", serviceRemote)
+        val serviceId = registerServiceInLocalDc("echo", serviceLocal)
 
         // then: local called
-        waitUntilEchoCalledThroughEnvoyResponds(echoContainer)
+        waitServiceOkAndFrom("echo", serviceLocal)
 
         // when: no local instances
-        deregisterService(localId)
+        deregisterService(serviceId)
 
         // then: remote called
-        waitUntilEchoCalledThroughEnvoyResponds(echoContainer2)
+        waitServiceOkAndFrom("echo", serviceRemote)
 
         // when: local instances again
-        registerServiceInLocalDc("echo")
+        registerServiceInLocalDc("echo", serviceLocal)
 
         // then: local called
-        waitUntilEchoCalledThroughEnvoyResponds(echoContainer)
+        waitServiceOkAndFrom("echo", serviceLocal)
     }
 
     @Test
     fun `latency between service registration in remote dc and being able to access it via envoy should be similar to envoy-control polling interval`() {
         // when
-        val latency = measureRegistrationToAccessLatency { name, target ->
-            registerServiceInRemoteCluster(name, target)
-        }
+        val latency = measureRegistrationToAccessLatency(
+            registerService = { name, target -> registerServiceInRemoteDc(name, target) },
+            readinessCheck = { name, target -> waitServiceOkAndFrom(name, target) }
+        )
 
         // then
         logger.info("remote dc latency: $latency")
 
         val tolerance = Duration.ofMillis(400) + stateSampleDuration
         val expectedMax = (pollingInterval + tolerance).toMillis()
-        assertThat(latency.max()).isLessThanOrEqualTo(expectedMax)
+        Assertions.assertThat(latency.max()).isLessThanOrEqualTo(expectedMax)
```

Comment:
why did you removed static imports for `assertions` and `await`

Reply:
by accident, will correct  it